### ISSUE TITLE
LocalGenerator won't generate if query string caching is disabled

### ISSUE THREAD

**Issue by webrgp** [putyourlightson/craft-blitz#735]
### Bug Report

When I change the configuration from `QUERY_STRINGS_DO_NOT_CACHE_URLS` to `QUERY_STRINGS_CACHE_URLS_AS_UNIQUE_PAGES` it generates.

[LocalGenerator](https://github.com/putyourlightson/craft-blitz/blob/328244c51a4f58575dfa4ee720fa5d0efa8bf0ea/src/drivers/generators/LocalGenerator.php#L58) pass `$siteUris` to BaseCacheGenerator's `getUrlsToGenerate` function, which adds the `token` query string to the urls by default

https://github.com/putyourlightson/craft-blitz/blob/328244c51a4f58575dfa4ee720fa5d0efa8bf0ea/src/drivers/generators/BaseCacheGenerator.php#L175-L184

Which in turn get stopped by the CacheRequestService:

https://github.com/putyourlightson/craft-blitz/blob/328244c51a4f58575dfa4ee720fa5d0efa8bf0ea/src/services/CacheRequestService.php#L253-L263

#### Application Info

- PHP version: 8.1.30
- Craft edition & version: Pro 4.11.5
- Database driver & version: MySQL 8.0.32

#### Installed Plugins

- Admin Bar: 4.3.0
- Amazon S3: 2.2.1
- Blitz: 4.23.5
- Blitz CloudFront Purger: 4.1.2
- Control Panel CSS: 2.6.0
- Cookies: 4.0.1
- CP Field Inspect: 1.4.4
- Element API: 4.1.0
- Expanded Singles: 2.0.5
- Google Custom Search: 2.0.4
- Google Maps: 4.5.2
- KeyChain: 4.0.0
- Logs: 4.0.1
- MatrixMate: 2.2.2
- Mix: v1.7.2
- Navigation: 2.1.0
- New Relic: 4.0.1
- Position Fieldtype: 4.0.0
- Redactor: 3.1.0
- Retour: 4.1.19
- SAML Service Provider: 4.3.0
- SEOmatic: 4.1.2
- Super Table: 3.0.15
- Typed link field: 2.1.5

#### Loaded Modules

- businesslogic: modules\businesslogic\BusinessLogicModule
- customization: modules\customization\Customization
- smart-recruiters: modules\smartrecruiters\SmartRecruiters
- closure: nystudio107\closure\Closure
- codeeditor: nystudio107\codeeditor\CodeEditor
- sprig-core: putyourlightson\sprig\Sprig
- blitz-hints: putyourlightson\blitzhints\BlitzHints
- verbb-base: verbb\base\Base

#### Blitz Plugin Settings

```json
{
    "debug": true,
    "hintsEnabled": true,
    "cachingEnabled": true,
    "refreshCacheEnabled": true,
    "refreshMode": 2,
    "includedUriPatterns": [
        {
            "enabled": true,
            "uriPattern": ".*"
        }
    ],
    "excludedUriPatterns": [
        {
            "uriPattern": "^search"
        },
        {
            "siteId": 2,
            "uriPattern": "^buscar"
        }
    ],
    "cacheStorageType": "putyourlightson\\blitz\\drivers\\storage\\FileStorage",
    "cacheStorageSettings": {
        "folderPath": "@webroot\/cache\/blitz",
        "createGzipFiles": false,
        "countCachedFiles": true
    },
    "cacheStorageTypes": [],
    "cacheGeneratorType": "putyourlightson\\blitz\\drivers\\generators\\LocalGenerator",
    "cacheGeneratorSettings": {
        "concurrency": 1
    },
    "cacheGeneratorTypes": [],
    "customSiteUris": [],
    "cachePurgerType": "putyourlightson\\blitz\\drivers\\purgers\\DummyPurger",
    "cachePurgerSettings": [],
    "cachePurgerTypes": [
        "putyourlightson\\blitzcloudfront\\CloudFrontPurger"
    ],
    "deployerType": "putyourlightson\\blitz\\drivers\\deployers\\DummyDeployer",
    "deployerSettings": [],
    "deployerTypes": [],
    "ssiEnabled": false,
    "ssiTagFormat": "<!--#include virtual=\"{uri}\" -->",
    "detectSsiEnabled": true,
    "esiEnabled": false,
    "queryStringCaching": 0,
    "includedQueryStringParams": [
        {
            "queryStringParam": ".*",
            "siteId": ""
        }
    ],
    "excludedQueryStringParams": [
        {
            "queryStringParam": "gclid",
            "siteId": ""
        },
        {
            "queryStringParam": "fbclid",
            "siteId": ""
        }
    ],
    "apiKey": "",
    "generatePagesWithQueryStringParams": true,
    "purgeAssetImagesWhenChanged": true,
    "refreshCacheAutomaticallyForGlobals": false,
    "refreshCacheWhenElementMovedInStructure": true,
    "refreshCacheWhenElementSavedUnchanged": false,
    "refreshCacheWhenElementSavedNotLive": false,
    "cacheNonHtmlResponses": false,
    "trackElements": true,
    "trackElementQueries": true,
    "excludedTrackedElementQueryParams": [],
    "cacheElements": true,
    "cacheElementQueries": true,
    "cacheDuration": null,
    "nonCacheableElementTypes": [],
    "sourceIdAttributes": [],
    "liveStatuses": [],
    "integrations": [
        "putyourlightson\\blitz\\drivers\\integrations\\SeomaticIntegration"
    ],
    "defaultCacheControlHeader": "no-store",
    "cacheControlHeader": "public, s-maxage=31536000, max-age=0",
    "cacheControlHeaderExpired": "public, s-maxage=5, max-age=0",
    "sendPoweredByHeader": false,
    "outputComments": false,
    "refreshCacheJobPriority": 10,
    "driverJobBatchSize": 10,
    "driverJobPriority": 100,
    "queueJobTtr": 300,
    "maxRetryAttempts": 10,
    "maxUriLength": 255,
    "mutexTimeout": 1,
    "commands": [],
    "injectScriptEvent": "DOMContentLoaded",
    "injectScriptPosition": 3
}
```

#### Recommendations

- ❌ The Blitz Hints utility is reporting 64 eager-loading opportunities.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but unchanged.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but not live.
- ✅ Image transforms are configured to be generated before page load.
- ✅ One or more globals exist and refreshCacheAutomaticallyForGlobals is disabled.
- ✅ The @web alias is explicitly defined.
- ✅ Queue jobs are configured not to run automatically via web requests.
- ✅ The Async Queue plugin is not installed or enabled.
- ✅ The blitz/cache/refresh-expired console command has been executed within the past 24 hours.

#### Site Tracking [1]

- Tracked Pages: 2
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 190
	- verbb\navigation\elements\Node: 82
	- craft\elements\Entry: 58
	- craft\elements\Asset: 41
	- craft\elements\Category: 9
- Tracked Element Queries: 226
	- verbb\navigation\elements\Node: 169
	- craft\elements\Entry: 48
	- craft\elements\Category: 9
- Tracked Tags: 0

#### Site Tracking [2]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [3]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [4]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [5]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [6]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [7]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [8]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 148
	- verbb\navigation\elements\Node: 83
	- craft\elements\Entry: 55
	- craft\elements\Category: 9
	- craft\elements\Asset: 1
- Tracked Element Queries: 223
	- verbb\navigation\elements\Node: 166
	- craft\elements\Entry: 48
	- craft\elements\Category: 9
- Tracked Tags: 0

#### Site Tracking [9]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 34
	- craft\elements\Entry: 13
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 31
	- verbb\navigation\elements\Node: 29
	- craft\elements\Entry: 2
- Tracked Tags: 1

#### Site Tracking [10]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 33
	- craft\elements\Entry: 12
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 32
	- verbb\navigation\elements\Node: 30
	- craft\elements\Entry: 2
- Tracked Tags: 0

#### Site Tracking [11]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 33
	- craft\elements\Entry: 12
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 32
	- verbb\navigation\elements\Node: 30
	- craft\elements\Entry: 2
- Tracked Tags: 0

#### Site Tracking [12]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 33
	- craft\elements\Entry: 12
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 32
	- verbb\navigation\elements\Node: 30
	- craft\elements\Entry: 2
- Tracked Tags: 0

#### Site Tracking [13]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 33
	- craft\elements\Entry: 12
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 32
	- verbb\navigation\elements\Node: 30
	- craft\elements\Entry: 2
- Tracked Tags: 0

#### Site Tracking [14]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 33
	- craft\elements\Entry: 12
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 32
	- verbb\navigation\elements\Node: 30
	- craft\elements\Entry: 2
- Tracked Tags: 0

#### Site Tracking [15]

- Tracked Pages: 1
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 33
	- craft\elements\Entry: 12
	- verbb\navigation\elements\Node: 12
	- craft\elements\Asset: 9
- Tracked Element Queries: 32
	- verbb\navigation\elements\Node: 30
	- craft\elements\Entry: 2
- Tracked Tags: 0

**Comment by bencroker** [comment]
As per https://github.com/putyourlightson/craft-blitz/issues/733#issuecomment-2474486277, can you please test the fix at df7ff76? You can do so by running `composer require "putyourlightson/craft-blitz:4.x-dev as 4.23.7"`.

**Comment by webrgp** [comment]
The fix works locally, and looking at the code, it seems it will work in prod too. Thanks Ben!

**Comment by bencroker** [comment]
Released in [4.23.7](https://github.com/putyourlightson/craft-blitz/releases/tag/4.23.7) and [5.9.6](https://github.com/putyourlightson/craft-blitz/releases/tag/5.9.6).
